detect_secrets/core/audit.py

```diff
@@ -451,63 +453,33 @@ def _get_secret_with_context(
 
     :raises: SecretNotFoundOnSpecifiedLineError
     """
-    secret_lineno = secret['line_number']
-    start_line = 1 if secret_lineno <= lines_of_context \
-        else secret_lineno - lines_of_context
-    end_line = secret_lineno + lines_of_context
-
-    output = subprocess.check_output([
-        'sed',
-        '-n', '{},{}p'.format(start_line, end_line),
-        filename,
-    ]).decode('utf-8').splitlines()
-
-    trailing_lines_of_context = lines_of_context
-    if len(output) < end_line - start_line + 1:
-        # This handles the case of a short file.
-        num_lines_in_file = int(subprocess.check_output([
-            # https://stackoverflow.com/a/38870057
-            # - 'wc -l' cannot be used here because if the last char
-            # of the file isn't \n, then the last line isn't counted
-            'grep',
-            '-c',
-            '',
-            filename,
-        ]).decode('utf-8').split()[0])
+    secret_line_idx = secret['line_number'] - 1
+    end_line = secret_line_idx + lines_of_context + 1
 
-        trailing_lines_of_context = lines_of_context - \
-            (end_line - num_lines_in_file)
+    if secret_line_idx <= lines_of_context:
+        start_line = 0
+        index_of_secret_in_output = secret_line_idx
+    else:
+        start_line = secret_line_idx - lines_of_context
+        index_of_secret_in_output = lines_of_context
 
-    # -1, because that's where the secret actually is (without it,
-    # it would just be the start of the context block).
-    # NOTE: index_of_secret_in_output should *always* be negative.
-    index_of_secret_in_output = -trailing_lines_of_context - 1
+    with codecs.open(filename, encoding='utf-8') as file:
+        output = list(itertools.islice(file.read().splitlines(), start_line, end_line))
 
-    try:
-        output[index_of_secret_in_output] = _highlight_secret(
-            output[index_of_secret_in_output],
-            secret_lineno,
-            secret,
-            filename,
-            plugin_settings,
-        )
-    except SecretNotFoundOnSpecifiedLineError:
```

Comment:
I believe `_highlight_secret` can still throw a `SecretNotFoundOnSpecifiedLineError`, in the case that the baseline is out-of-date.

So I think we do need to keep the `try:` `exception:` code here https://github.com/Yelp/detect-secrets/blob/f0ece437079854910a97394451b3ab66f99ae515/detect_secrets/core/audit.py#L486-L503

Reply:
It's updated in the latest commit https://github.com/Yelp/detect-secrets/pull/120/commits/4d5ab73816d23081ef1e8b865739db8c22f99c1f